File: response.cpp
#include "HotelManager.h"

void Response::setErrorMessage(std::string errorMessage) {
	isSuccess = false;
	_errorMessage = errorMessage;
}

std::string Response::getErrorMessage() {
	return _errorMessage;
}

void Response::setResult(Query &result) {
	isSuccess = true;
	_result = *(new Query(result));
}

Query Response::getResult() {
	return *(new Query(_result));
}

void Response::setOrderList(std::vector<Query> &orderList) {
	isSuccess = true;
	_orderList = orderList;
}

void Response::setHotelInfoList(std::vector<HotelInfo> &hotelInfoList) {
	isSuccess = true;
	_hotelInfoList = hotelInfoList;
}

File: Client/src/Response.java
import java.util.List;

public class Response implements java.io.Serializable
{
	private static final long serialVersionUID = 1L;
	
	public boolean isSuccess;
	public String errorMessage;
	public Query result;
	public List<Query> orderList;
	public List<HotelInfo> hotelInfoList;
	/**
	 * 12/20 add historyList
	 */
	public List<Query> historyList;
	
	public void setErrorMessage(String errorMessage) {
		isSuccess = false;
		this.errorMessage = errorMessage;
	}
	
	public String getErrorMessage() {
		return errorMessage;
	}
	
	public void setResult(Query result) {
		isSuccess = true;
		this.result = result.clone();
	}
	
	public Query getResult() {
		return result.clone();
	}
	
	public void setOrderList(List<Query> orderList) {
		isSuccess = true;
		this.orderList = orderList;
	}
	
	public void setHotelInfoList(List<HotelInfo> hotelInfoList) {
		isSuccess = true;
		this.hotelInfoList = hotelInfoList;
	}
	/**
	 * 12/20 add setHistoryList
	 * @param historyList
	 */
	public void setHistoryList(List<Query> historyList) {
		isSuccess = true;
		this.historyList  = historyList;
	}
}
